File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    deploy = []
    count = 0
    
    for i in range(len(progresses)):
        rest = (100-progresses[i]) % speeds[i] 
        progress = (100-progresses[i]) // speeds[i] 
        if rest == 0:
            deploy.append(progress)
            continue
        deploy.append(progress+1)
    
    while(len(deploy) > 0):
        count = 0
        
        for i in range(len(deploy)):
            deploy[i] -= 1
            
        while(len(deploy) > 0 and deploy[0] <= 0):
            deploy.pop(0)
            count += 1
        if count != 0:
            answer.append(count)
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public List<Integer> solution(int[] progresses, int[] speeds) {
        
        // 0. 준비
        List<Integer> answer = new ArrayList<>();
        Queue<Integer> pq = new LinkedList<>();
        Queue<Integer> tq = new LinkedList<>();
        
        for(int i = 0; i<progresses.length; i++){
            pq.add(progresses[i]);
            tq.add(speeds[i]);
        }
        
    
        // 1. 탐색
        while(!pq.isEmpty()) {
            int progress = pq.poll();
            double time = tq.poll();
            
            // 남은 작업량 계산
            int job = (int) Math.ceil((100 - progress) / time);
            //System.out.println(Math.ceil(0.01));
            // 배포 가능한 기능 계산
            int count = 1;
            while(!pq.isEmpty()){
                //System.out.println((int) Math.ceil((100 - pq.peek()) / tq.peek()));
                if(job < (int) Math.ceil((double)(100 - pq.peek()) / tq.peek())) {
                    break;
                }
                pq.poll();
                tq.poll();
                count++;
            }
            answer.add(count);
        }
            
        // 2. 정답 출력
        return answer;
    }
}
